### DISCUSSION TITLE
Import Collections

### DISCUSSION THREAD

**Discussion by hlanderdev** [pocketbase/pocketbase#938] [Q&A] [answered]
So, I have a small custom command for importing and exporting collections via JSON, that I am interested as putting in as a PR to v0.8.0. Currently, there is a "migrate collections" command that exports a Go file that is nice if you are working with Pocketbase as a framework (which I do), but when it is already compiled it has limited use. I think it would be more versatile if it exported only JSON and then could import a JSON file with a separate command (for example, "migrate collections export" and "migrate collections import"). This is how my custom commands currently implements it. I guess my questions is: before I go through the trouble of putting in a PR how would you like something like that to be implemented?

**Comment by ganigeorgiev** [comment, marked as answer]
I'm planning to handle this after the v0.8.0 release in relation to https://github.com/pocketbase/pocketbase/discussions/489.

My idea was to add support for writing migrations via a JS interpreter using similar api to the Go one because we'll be able to handle the case where the user want to apply a db change different than a collection import/export, eg. setting a default value for newly added field, running an import script or executing something completely different that need to be done only once. Or in other words we'll have 2 migration commands - the default Go one available when used as framework and the JS interpreter version for the prebuilt binary.

_@benallfree already started experimenting with [goja](https://github.com/dop251/goja) in https://github.com/benallfree/pockethost and we can do something similar but to keep it simple for now I was thinking to focus only on migrations and later we may add also support for the other Go apis._

**Reply by hlanderdev** [reply]
I'm glad I asked then. It seems you are planning something a lot more robust than my solution. I'm curious how that will look when it comes out. Thanks!
